### DISCUSSION TITLE
Any plans for supporting NCZarr format?

### DISCUSSION THREAD

**Discussion by mliukis** [pydata/xarray#5638] [General] [answered]
Just wondering if support for NCZarr format is in consideration for the xarray.
Thanks!

**Comment by shoyer** [comment, marked as answer]
You _should_ be able to read NCZarr files via `engine='netcdf4'` in `xarray.open_dataset`, which uses the underlying netCDF C library in which NCZarr is implemented. But I don't know if anyone has tested it yet.

We have not yet implemented NCZarr in `xarray.open_zarr` (via the Zarr Python library). This is something we would be interested in supporting in principle, but it hasn't been implemented yet. There are also still some unresolved questions about how to reconcile the NCZarr convention and what we currently do in Xarray: https://github.com/zarr-developers/zarr-specs/issues/41

**Reply by mliukis** [reply]
Is there a way to write xarray.Dataset to NCZarr?

**Reply by shoyer** [reply]
Again, it's _probably_ possible to write to NCZarr via `to_netcdf()` with `engine='netcdf4'` if you supply an appropriate URL: https://www.unidata.ucar.edu/blogs/developer/entry/overview-of-zarr-support-in
